### PR TITLE
restored "Publishers with most sales" section

### LINKED ISSUE #1522: Put back "Publishers with most sales" section to front page
We had this in v3, we didn't get to it in v4. Goal is to add it back but most likely needs big refactor to get the numbers, like making use of what Aquarius gives us as described in #1510

### PR DESCRIPTION
**Pull request by EnzoVezzaro**
Fixes #1522 .

Changes proposed in this PR:

- restored "Publishers with most sales" section in homepage

### CODE REVIEW

File: src/components/Home/index.tsx [hunk 1]
```diff
@@ -99,6 +102,8 @@ function SectionQueryResult({
         assets={result?.results}
         showPagination={false}
         isLoading={loading || !query}
+        chainIds={chainIds}
```
[the remark this answers is not in the document]
Comment (EnzoVezzaro) [reply src/components/Home/index.tsx:105]: Yeah, probably there's something wrong on my IDE. 

Screen Shot 2022-06-23 at 9 08 04 AM

But I checked the correct interface, you're right. Will remove it 👍🏽

Screen Shot 2022-06-23 at 9 10 08 AM

[the remark this answers is not in the document]
Comment (kremalicious) [reply src/components/Home/index.tsx]: how fun is that? But yeah, I run into this in VS Code when switching between multiple branches and editor sometimes can't keep up with updating all typings again for newly switched branch

**Comment by kremalicious** [comment]
Then for the data, this seems not so correct. Why has [this account](https://market-git-feature-issue-1522-most-sales-section-oceanprotocol.vercel.app/profile/0x9bf750b5465a51689fa4235aac1f37ec692ef7b4) 55 sales listed and shows up first, but has no assets published?

**Review by claudiaHash — COMMENTED** [review]
Yeah,  when selecting all the test networks I get this:

Screenshot from 2022-06-23 16-22-01

**Comment by EnzoVezzaro** [comment]
> Then for the data, this seems not so correct. Why has [this account](https://market-git-feature-issue-1522-most-sales-section-oceanprotocol.vercel.app/profile/0x9bf750b5465a51689fa4235aac1f37ec692ef7b4) 55 sales listed and shows up first, but has no assets published?

Mmmm... very interesting. 🧐 looking into it

Screen Shot 2022-06-23 at 9 43 56 AM

**Comment by kremalicious** [comment]
> There you go [mystery solved](https://v4.subgraph.polygon.oceanprotocol.com/subgraphs/name/oceanprotocol/ocean-subgraph/graphql?query=%7B%0A%20%20user(id%3A%220x9bf750b5465a51689fa4235aac1f37ec692ef7b4%22)%7B%0A%20%20%20%20id%0A%20%20%20%20totalOrders%0A%20%20%20%20%0A%20%20%20%20totalSales%0A%20%20%7D%2C%0A%20%20nfts(where%3A%7Bowner%3A%220x9bf750b5465a51689fa4235aac1f37ec692ef7b4%22%7D)%7B%0A%20%20%20%20id%2C%0A%20%20%20%20providerUrl%0A%7D%0A%7D)

good point. So we need to do something with this information as we can't just list them and then users can't find any assets for them. The point of this section is mainly for liquidity providers to figure out where highest consume volume is to be expected. So listing publishers from other marketplaces does not make sense. So we need to figure out a way for this section on front page to only include publishers and their assets in the market interface users are looking at.

Additionally I don't think it makes sense including any sales on test networks at all, so section should only include sales in main networks, but they then still should be attached to network selector to include those which are selected.

**Comment by mihaisc** [comment]
The component should behave like the highest liquidity. If you select a test network it will display top sellers for that network. Only the footer is hardcoded for main networks. 
Idea to only do this for stuff in our market  https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-sum-aggregation.html

**Comment by EnzoVezzaro** [comment]
> The component should behave like the highest liquidity. If you select a test network it will display top sellers for that network. Only the footer is hardcoded for main networks. Idea to only do this for stuff in our market https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-sum-aggregation.html

Thank you for the review guys. As for the network selection, right now we're basically doing what our profile page does: it [fetches the `top 20`  results filtered by  `totalSales`](https://github.com/oceanprotocol/market/blob/feature/issue-1522-most-sales-section/src/@utils/subgraph.ts#L416:L416) on the selected networks. So, it will get the most updated `totalSales` for the user on all selected networks. 

As for the second matter, basically we need to remove from the total count all assets published outside our main market, correct? We can do this by checking `providerUrl` results (for example if contains `oceanprotocol.com`). I have some questions though, In this [specific case](https://github.com/oceanprotocol/market/pull/1540#issuecomment-1164526258), we have `totalSales` = 55, `totalOrders` = 24, and the `nft` results gives me 17 assets (which I imagine is because we have multiple orders for the same `nft`, but checking the `orders` they all have an amount of '1'). Also, where's or what's the difference between `totalSales` - `totalOrders` = 31? I'd expect these two numbers to aligned. Probably I'm misunderstanding something here.

**Comment by EnzoVezzaro** [comment]
Changed my implementation to get the total sales with `aggregations` as [suggested](https://github.com/oceanprotocol/market/pull/1540#issuecomment-1166223016) from ES.

Now, it seems we have different values between homepage and profile page. 

Homepage

Screenshot 2022-07-04 at 07 51 52

Profile

Screenshot 2022-07-04 at 07 52 24

I wonder if what I tried here is actually what we need, and if so, we'll need to move this logic to profile too.

**Comment by mihaisc** [comment]
We need to move the logic in the profile as well. We need to be consistent. But why are they different for Winston, this is weird.

**Comment by EnzoVezzaro** [comment]
> We need to move the logic in the profile as well. We need to be consistent. But why are they different for Winston, this is weird.
> 
> It seems that it's just spamming requests thus being very slow image

Not sure why Winston is different, but I guest is related to this `null` values in provider

Screenshot 2022-07-04 at 09 47 38

as for the request, probably I need to change the way we get the user's list [here](https://github.com/oceanprotocol/market/blob/feature/issue-1522-most-sales-section/src/@utils/subgraph.ts#L442). Right now we're first fetching the top publishers for each chain (so if all networks are active: it's 4 mainets and 4 testnets) grouped by user, and then we're fetching the correct value from `aquarius` to show in the UI. So when all networks are active, this query takes a really long time to do all. I'm making some changes to see if we can speed it up.

**Comment by mihaisc** [comment]
You need just one query, something like 
```
{
    "aggs": {
        "a1": {
            "terms": {
                "field": "nft.owner.keyword",
                "order": {"totalSales": "desc"}
            },
            "aggs": {
                "totalSales": {
                    "sum": {
                        "field": "stats.orders"
                    }
                }
            }
        }
    }
}
``` 
returns something like : 
```
 {
                    "doc_count": 2,
                    "key": "0xbbd33AFa85539fa65cc08A2e61a001876D2f13FE",
                    "totalSales": {
                        "value": 50.0
                    }
                },
                {
                    "doc_count": 16,
                    "key": "0x903322C7E45A60d7c8C3EA236c5beA9Af86310c7",
                    "totalSales": {
                        "value": 44.0
                    }
                },
...
```

**Comment by kremalicious** [comment]
this is great, excellent request-optimizing vibe in here and this section now loads quicker than all the others! 

Did some small typography tweaking and markup cleanup so that our `DotDotDot` component actually works now for long publisher names:

Screen Shot 2022-07-05 at 13 31 19

**Review by kremalicious — APPROVED** [review]

**Pull request merged**